src/qtism/runtime/tests/Utils.php

```diff
@@ -86,13 +89,24 @@ public static function isResponseValid(QtiDatatype $response = null, ResponseVal
 
             $patternMask = OperatorUtils::prepareXsdPatternForPcre($patternMask);
 
+            $isMaxEntryRestriction = preg_match(self::MAX_ENTRY_RESTRICTION_PATTERN, $patternMask, $matches)
+                && self::isSingleMatchGroup($patternMask);
+
             foreach ($values as $value) {
-                $result = @preg_match($patternMask, $value);
+                if ($isMaxEntryRestriction) {
+                    extract($matches);
+                    $entries = count(preg_split("/$splitPattern/", $response)) - 1;
+                    if ($entries > $max || $entries < $min) {
+                        return false;
+                    }
+                } else {
+                    $result = @preg_match($patternMask, $value);
```

Comment:
WDYT we try regular `preg_match()` first, and then if failed with an error, try the new approach?
I think the regular preg_match() covers the majority of the cases, and therefore should be default.